tests/pkg-layering/main.yml

```
@@ -0,0 +1,319 @@
+      set_fact:
+        pending_pkgs: "{{ pending_json['deployments'][0]['packages'] }}"
+      when: pending_json['deployments'][0] is defined and not pending_json['deployments'][0]['booted']
+
+    - name: Set packages when deployment 1 is not booted
+      set_fact:
+        pending_pkgs: "{{ pending_json['deployments'][1]['packages'] }}"
+      when: pending_json['deployments'][1] is defined and not pending_json['deployments'][1]['booted']
+
+    - name: Fail if {{ g_pkg2 }} is in rpm-ostree status output for pending packages
+      fail:
+        msg: "{{ g_pkg2 }} in rpm-ostree status output"
+      when: "'{{ g_pkg2 }}' in pending_pkgs"
+
+    - name: Fail if {{ g_pkg1 }} is not in rpm-ostree status output for pending packages
+      fail:
+        msg: "{{ g_pkg1 }} not in rpm-ostree status output"
+      when: "'{{ g_pkg1 }}' not in pending_pkgs"
+
+    - name: Fail if {{ g_pkg1 }} is installed
+      command: command -v {{ g_pkg1 }}
+      register: binary
+      failed_when: binary.rc != 1
+
+    # Verify that after additional repos are enabled (i.e. EPEL), rpm-ostree install|uninstall can be successfully used for packages in those repos.
+    - name: Retrieve EPEL rpm
+      get_url:
+        url: https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+        dest: /tmp
+        mode: 0600
+
+    - name: Extract GPG Key to install directory
+      shell: >
+        rpm2cpio /tmp/epel-release-latest-7.noarch.rpm |
+        cpio -iv --to-stdout ./etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7 >
+        /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
+
+    - name: Extract EPEL repo file to install directory
+      shell: >
+        rpm2cpio /tmp/epel-release-latest-7.noarch.rpm |
+        cpio -iv --to-stdout ./etc/yum.repos.d/epel.repo >
+        /etc/yum.repos.d/epel.repo
+
+    - name: Install {{ g_epel_pkg }} from EPEL
+      command: rpm-ostree install {{ g_epel_pkg }} -r
+      async: 60
+      poll: 0
+      ignore_errors: true
```

Hm, but we don't want to ignore errors...let's do the reboot as a separate step?